### ISSUE TITLE
104: Connection reset by peer

### ISSUE THREAD

**Issue by zjhzxhz** [phalcon/cphalcon#11229]
When I'm using Phalcon 2.0.9 with Nginx, I got an error message:

```
[error] 525#0: *4 recv() failed (104: Connection reset by peer) while reading response header from upstream, client: *.*.*.*, server: localhost, request: "GET /TestZilla/public/index.php HTTP/1.1", upstream: "fastcgi://127.0.0.1:9000", host: "*.*.*.*"
```

And the Laravel framework works well.

BTW I think I have installed the Phalcon framework correctly. Because I got following output using `print_r(get_loaded_extensions())`:

```
Array ( 
	[0] => Core 
	[1] => date 
	[2] => ereg 
	[3] => libxml 
	[4] => openssl 
	[5] => pcre 
	[6] => sqlite3 
	[7] => zlib 
	[8] => bcmath 
	[9] => bz2 
	[10] => calendar 
	[11] => ctype 
	[12] => curl 
	[13] => dba 
	[14] => dom 
	[15] => hash 
	[16] => fileinfo 
	[17] => filter 
	[18] => ftp 
	[19] => gd 
	[20] => gettext 
	[21] => gmp 
	[22] => SPL 
	[23] => iconv 
	[24] => json 
	[25] => mbstring 
	[26] => mcrypt 
	[27] => session 
	[28] => mysql 
	[29] => mysqli 
	[30] => standard 
	[31] => PDO 
	[32] => mysqlnd 
	[33] => pdo_sqlite 
	[34] => Phar 
	[35] => posix 
	[36] => readline 
	[37] => Reflection 
	[38] => pdo_mysql 
	[39] => shmop 
	[40] => SimpleXML 
	[41] => soap 
	[42] => sockets 
	[43] => exif 
	[44] => sysvmsg 
	[45] => sysvsem 
	[46] => sysvshm 
	[47] => tokenizer 
	[48] => wddx 
	[49] => xml 
	[50] => xmlreader 
	[51] => xmlwriter 
	[52] => xsl 
	[53] => zip 
	[54] => cgi-fcgi 
	[55] => phalcon 
	[56] => mhash 
)
```

Environment:

- PHP 5.6.16
- [Tengine 2.1.1](http://tengine.taobao.org)

Who can help me to solve this problem?

**Comment by zjhzxhz** [comment]
I found that this error is caused by `getContent()`:

```
$application = new Application($di);
echo $application->handle()->getContent();
```

Everything goes well until execute `echo $application->handle()->getContent();`.

But I don't know how to solve it :(.

**Comment by daison12006013** [comment]
@zjhzxhz, try this nginx settings https://github.com/phalconslayer/slayer/blob/dev/storage/etc/nginx/sites-available/slayer.app, it might be your nginx causes the problem.

while just searching a bit about the error you have, is the same like this http://serverfault.com/questions/543999/nginx-errors-recv-failed-104-connection-reset-by-peer-while-reading-respon

**Comment by zjhzxhz** [comment]
@daison12006013 I have to say sorry that I missed one important thing.
Actually I have solved this problem.

I ran the script `build/install.sh` on a Server with 2 GB memory (with the same environment). 
And I packed the `/tmp/cphalcon` folder to cphalcon.tar.gz.
And I unpacked the compressed file to another VPS with 1 GB memory. And execute `make install` on that server.
I did this because I couldn't run the script on this VPS with 1GB memory.

And I enlarge this VPS to 2GB memory and reran `build/install.sh`. And everything goes well now.
Any way, thanks a lot.
